### ISSUE TITLE
The test creates token_ids with values from 0 to 1000, but indices 3 and 4 are padding (value 0 in the first sequence, indices 2-4 in the second). The assertion checks indices 3 onwards in both sequences, but this doesn't align with where the actual padding is. The test should use explicit padding token ID (0) in the input and check those specific positions, not assume padding based on position.

### ISSUE THREAD

**Issue by tzervas** [tzervas/tritter#24]
The test creates token_ids with values from 0 to 1000, but indices 3 and 4 are padding (value 0 in the first sequence, indices 2-4 in the second). The assertion checks indices 3 onwards in both sequences, but this doesn't align with where the actual padding is. The test should use explicit padding token ID (0) in the input and check those specific positions, not assume padding based on position.
```suggestion
        # Padding embeddings should be zero at positions where token_ids == padding_idx (0)
        padding_mask = token_ids == 0
        padding_embeddings = output[padding_mask]
        assert torch.allclose(padding_embeddings, torch.zeros_like(padding_embeddings))
```

_Originally posted by @copilot-pull-request-reviewer in https://github.com/tzervas/tritter/pull/2#discussion_r2713634032_
